ydiff.py

```diff
@@ -743,10 +743,20 @@ def markup_to_pager(stream, opts):
     See issue #30 (https://github.com/ymattw/ydiff/issues/30) for more
     information.
     """
-    pager_cmd = ['less']
-    if not os.getenv('LESS'):
-        # Args stolen from git source: github.com/git/git/blob/master/pager.c
-        pager_cmd.extend(['-FRSX', '--shift 1'])
+    pager_cmd = [opts.pager]
+    pager_opts = (opts.pager_options.split(' ')
+                  if opts.pager_options is not None
+                  else None)
+
+    if (opts.pager.lower() == 'less' and
+            not os.getenv('LESS') and
```

Comment:
I understand you want to make it work for Windows, but what if user provides --pager=/usr/bin/less?

If we let --pager default to None, here we can use:

```
if opts.pager is None:
    pager_cmd = ['less']
    if not os.getenv('LESS') and not opts.pager_options:
        pager_cmd.extend(['-FRSX', '--shift 1'])
````

This way we only keep the original behavior when neither --pager nor --pager-options is explicitly defined.

Reply:
My changes really had nothing to do with Windows.

So by going through this logic, in effect you are defaulting the pager to `less` anyway. I just bubbled that up to the options section. But if you'd rather me do it this way, I can. I was trying to make your default `less` logic work the way it did before, with the addition of the ability for it to work using `pager_options` as per https://github.com/ymattw/ydiff/pull/87#issuecomment-544246504

Reply:
I thought the tolower() thing is for Windows. And I did not mention in the original comment, but defaulting to less is slightly a behavior change:

1. Before
   a. (Pager not set) Use `less` AND set default `LESS` environment
2. After
   a. When pager not set, default to `less` and **NOT** set default `LESS`
   b. When pager is set, use the specified pager
   c. When pager is set, when it's `less` and pager-option is not set, set default pager options

Now 1a and 2a is inconsistent. I am just worry about this behavior change.

Also 2c is what we want to avoid, let's not set default options for users. This is the issue about - the "sensitive default" I thought turns out to be annoying for some users.  Checking for the name `less` is not necessary, the `LESS` environment variable will be respected by `less` itself anyway.

I hope it makes sense. Thanks again, and looking forward to your feedback (and the README update :).

Reply:
Thanks. I think I understand where you're coming from now.